monai/transforms/post/array.py

```diff
@@ -183,52 +184,90 @@ class KeepLargestConnectedComponent(Transform):
     """
 
     def __init__(
-        self, applied_values, independent: bool = True, background: int = 0, connectivity: Optional[int] = None
+        self,
+        applied_values=None,
+        independent: bool = True,
+        background: int = 0,
+        connectivity: Optional[int] = None,
+        applied_channel_indices=None,
     ):
         """
         Args:
             applied_values (list or tuple of int): number list for applying the connected component on.
-                The pixel whose value is not in this list will remain unchanged.
+                The pixel whose value is not in this list will remain unchanged. This will be used only
+                if the data has one channel.
             independent (bool): consider several labels as a whole or independent, default is `True`.
                 Example use case would be segment label 1 is liver and label 2 is liver tumor, in that case
                 you want this "independent" to be specified as False.
             background: Background pixel value. The over-segmented pixels will be set as this value.
             connectivity: Maximum number of orthogonal hops to consider a pixel/voxel as a neighbor.
                 Accepted values are ranging from  1 to input.ndim. If ``None``, a full
                 connectivity of ``input.ndim`` is used.
+            applied_channel_indices (list): If the data is in one-hot format, this is used to determine what channels
+                to apply.
         """
         super().__init__()
         self.applied_values = applied_values
         self.independent = independent
         self.background = background
         self.connectivity = connectivity
-        if background in applied_values:
-            raise ValueError("Background pixel can't be in applied_values.")
+        self.applied_channel_indices = applied_channel_indices
+
+        if applied_values is None and applied_channel_indices is None:
+            raise ValueError("Please provide either applied_values or applied_channel_indices.")
+
+        if applied_channel_indices is None:
+            if background in applied_values:
+                raise ValueError("Background pixel can't be in applied_values.")
 
     def __call__(self, img):
         """
         Args:
-            img: shape must be (batch_size, 1, spatial_dim1[, spatial_dim2, ...]).
+            img: shape must be (batch_size, C, spatial_dim1[, spatial_dim2, ...]).
 
         Returns:
-            A PyTorch Tensor with shape (batch_size, 1, spatial_dim1[, spatial_dim2, ...]).
+            A PyTorch Tensor with shape (batch_size, C, spatial_dim1[, spatial_dim2, ...]).
         """
         channel_dim = 1
         if img.shape[channel_dim] == 1:
+            if self.applied_values is None:
```

Comment:
I think you don't need to check conditions by channel dim, because `applied_channel_indices=0` can also work for 1 channel data with [0, 1] values.
Instead, you can just use:
```py
if applied_values is not None:
    XXX
else:
    XXX
```
What do you think?